als-common/shared/src/main/scala/org/mulesoft/amfintegration/amfconfiguration/AmfConfigurationWrapper.scala

```diff
@@ -211,3 +194,62 @@ object AmfConfigurationWrapper {
       Future.successful(new Content(content, fileUrl))
   }
 }
+
+case class AMFConfigurationStateManager(validators: Seq[AMFShapePayloadValidationPlugin],
+                                        syntaxes: Seq[AMFSyntaxParsePlugin],
+                                        dialects: Seq[Dialect],
+                                        resourceLoaders: Seq[ResourceLoader]) {
+
+  def withValidators(v: Seq[AMFShapePayloadValidationPlugin]): AMFConfigurationStateManager =
+    AMFConfigurationStateManager(validators ++ v, syntaxes, dialects, resourceLoaders)
+
+  def withSyntaxes(s: Seq[AMFSyntaxParsePlugin]): AMFConfigurationStateManager =
+    AMFConfigurationStateManager(validators, syntaxes ++ s, dialects, resourceLoaders)
+
+  def withDialects(d: Seq[Dialect]): AMFConfigurationStateManager =
+    AMFConfigurationStateManager(validators, syntaxes, dialects ++ d, resourceLoaders)
```

Comment:
Currently there is no way to remove/replace a dialect?

Reply:
replacing should work by adding a dialect with the same ID or name/version, but I don't think removing dialects is part of the plan

Reply:
LGTM then